### PR TITLE
Add info to support cloud config

### PR DESCRIPTION
**Pull request by dedemorton**
Summary of changes:

- Added info to help cloud users get Beats set up correctly.
- Added more platform-specific commands for setting up and running Filebeat modules (to make the docs consistent).
- Did some minor cleanup to change product names > attribute references (still need to do a full cleanup, but trying to catch these as I open files).
- Moved the content about changing the output codec so that it appears at the end of the output config section (since the section applies to multiple outputs).
- Misc changes to make steps consistent, use similar language, etc across content.

I'm marking this as "in progress" because I need some input from reviewers before this is ready to merge. Please see comments flagged for REVIEWERS.

### CODE REVIEW

**Comment by dedemorton** [comment]
@exekias @ruflin Probably only needs one developer to review, so one of you is option. You get to decide who. :-)

File: filebeat/docs/include/running-modules.asciidoc [hunk 1]
```diff
@@ -2,54 +2,119 @@
 [id="running-{modulename}-modules"]
 === Set up and run the module
 
-IMPORTANT: If you’ve secured Elasticsearch and Kibana, you need to configure the
-`username` and `password` options in the Elasticsearch output before setting up
+// REVIEWERS: Should we include docker commands here, too?
```
Comment (exekias) [inline filebeat/docs/include/running-modules.asciidoc]: I don't think we should, as it is not straightforward in the docker world to follow these interactive steps

File: filebeat/docs/modules-getting-started.asciidoc [hunk 2]
```diff
@@ -1,141 +1,238 @@
...
+related to file ownership or permissions when you try to run {beatname_uc} modules.
 See {libbeat}/config-file-permissions.html[Config File Ownership and Permissions]
 in the _Beats Platform Reference_ if you encounter errors related to file
-ownership or permissions.
-
-This getting started guide uses the `--modules` flag to enable modules
-at the command line when you run Filebeat. In a production environment, you'll
-probably want to use the configs in the `modules.d` directory instead. See
-<<configuration-filebeat-modules>> for more information.
+ownership or permissions.
```
Comment (exekias) [inline filebeat/docs/modules-getting-started.asciidoc]: Both the note and the paragraph are talking about `errors related to file ownership or permissions`. May this sound repetitive? I'm not sure here, so it's your call :)

File: filebeat/docs/modules-getting-started.asciidoc [hunk 3]
```diff
@@ -1,141 +1,238 @@
...
 files. For example, the following command sets the path for the Nginx access
-logs:
+logs. For this command to work, the `nginx` module must be enabled.
 
-[source,shell]
+// REVIEWERS: I want to provide platform-specific examples, but I'm not sure
+// what theses commands look like for deb/rpm/windows. I assume we want to
+// tell users to start the service, but I don't think options like -M are
+// allowed when you start as a service. Are they? Hoping someone with
+// deb/rpm/windows experience can help me get the syntax right here.
```
Comment (exekias) [inline filebeat/docs/modules-getting-started.asciidoc]: There is no way to pass these variables to services, they would need to update the config file with them

Comment (dedemorton) [reply filebeat/docs/modules-getting-started.asciidoc]: @exekias Then I should probably add a note to indicate that. But I think users can still run the following commands, right?

deb and rpm:
```
filebeat -e -M "nginx.access.var.paths=[/var/log/nginx/access.log*]"
```

win:
```
PS > .\filebeat.exe -e -M "nginx.access.var.paths=[c:/programdata/nginx/logs/*access.log*]"
```

**Comment by tbragin** [comment]
cc @dov0211

File: filebeat/docs/modules-getting-started.asciidoc (hunk 3 above)
Comment (exekias) [reply filebeat/docs/modules-getting-started.asciidoc]: Those look correct :+1: but they are not useful to keep the service running, but for testing.

**Comment by dedemorton** [comment]
I've added the fixes from the review. I did a lot of refactoring in the Filebeat docs to remove duplicated content. You can see what the changes will look like here: https://filebeatreference-6af8e.firebaseapp.com/

Now all the commands in the getting started and reference content should show platform-specific commands. 

@exekias Changes are ready for a final review.

**Comment by dedemorton** [comment]
jenkins retest this please

**Comment by dedemorton** [comment]
@karenzone FYI...

**Pull request merged**